### PR TITLE
[Enhancement] Support Kerberos Authentication in RBAC

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 87 (05.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/mysql/MysqlHandshakePacket.java|0|11|00.00%|[163, 164, 165, 166, 167, 169, 170, 171, 172, 173, 174]|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|0|22|00.00%|[201, 202, 204, 205, 206, 207, 208, 209, 211, 216, 218, 219, 220, 221, 222, 223, 224, 226, 227, 228, 239, 240]|
|:large_blue_circle:|com/starrocks/authentication/KerberosAuthenticationProvider.java|2|27|07.41%|[31, 32, 35, 36, 37, 38, 40, 42, 49, 51, 52, 54, 58, 59, 61, 62, 63, 64, 65, 70, 71, 72, 73, 74, 75]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|3|27|11.11%|[576, 577, 578, 581, 582, 583, 586, 587, 588, 591, 593, 594, 595, 596, 598, 600, 602, 604, 606, 607, 608, 609, 612, 616]|

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java [hunk 1]
```diff
@@ -197,20 +197,36 @@ public static boolean negotiate(ConnectContext context) throws IOException {
             // 1. clear the serializer
             serializer.reset();
             // 2. build the auth switch request and send to the client
-            // TODO(yiming): support kerberos in new RBAC privilege framework later
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java:200]: There are other two todo marks, plz check whether we need to modify those logic, if not, plz remove them directly.

image

File: fe/fe-core/src/main/java/com/starrocks/authentication/KerberosAuthenticationProvider.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                throw new AuthenticationException("Failed to authenticate for [user: " + user + "] by kerberos");
+            }
+        } catch (Exception e) {
+            throw new AuthenticationException("Failed to authenticate for [user: " + user + "] " +
+                    "by kerberos, msg: " + e.getMessage());
+        }
+    }
+
+    @Override
+    public UserAuthenticationInfo upgradedFromPassword(UserIdentity userIdentity, Password password)
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/authentication/KerberosAuthenticationProvider.java:68]: This upgrade process is critical for our enterprise user, have we tested it manually?

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/authentication/KerberosAuthenticationProvider.java (hunk 2 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/authentication/KerberosAuthenticationProvider.java:68]: yes

**Review by nshangyiming — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**